catalogue/webapp/components/Calendar/Calendar.tsx

```diff
@@ -0,0 +1,278 @@
+              setTabbableDate(newMonth);
+              setPreviousMonthDisabled(
+                newMonth.clone().subtract(1, 'month').isBefore(min, 'month')
+              );
+              setNextMonthDisabled(
+                newMonth.clone().add(1, 'month').isAfter(max, 'month')
+              );
+            }}
+          >
+            previous
+          </button>
+          <button
+            type="button"
+            className="next-month"
+            aria-label="next month"
+            disabled={nextMonthDisabled}
+            onClick={() => {
+              const newMonth = tabbableDate.clone().add(1, 'month');
+              setTabbableDate(newMonth);
+              setPreviousMonthDisabled(
+                newMonth.clone().subtract(1, 'month').isBefore(min, 'month')
+              );
+              setNextMonthDisabled(
+                newMonth.clone().add(1, 'month').isAfter(max, 'month')
+              );
+            }}
+          >
+            next
+          </button>
+        </div>
+      </Header>
+      <Table
+        className={classNames({
+          [font('hnb', 6)]: true,
+        })}
+        role="grid"
+        aria-labelledby="id-grid-label"
+        onKeyDown={event => {
+          handleKeyDown(
+            event,
+            tabbableDate,
+            min,
+            max,
+            setTabbableDate,
+            setChosenDate
+          );
+        }}
+      >
+        <thead>
+          <tr>
+            <th scope="col" abbr="Monday">
+              M
+            </th>
+            <th scope="col" abbr="Tuesday">
+              T
+            </th>
+            <th scope="col" abbr="Wednesday">
+              W
+            </th>
+            <th scope="col" abbr="Thursday">
+              T
+            </th>
+            <th scope="col" abbr="Friday">
+              F
+            </th>
+            <th scope="col" abbr="Saturday">
+              S
+            </th>
+            <th scope="col" abbr="Sunday">
+              S
+            </th>
+          </tr>
+        </thead>
+        <tbody>
+          {rows.map((row, i) => {
+            return (
+              <tr key={i}>
+                {row.map((date, i) => {
+                  const numberOfDaysInMonth = tabbableDate.daysInMonth();
+                  const closestTabbableDate =
+                    numberOfDaysInMonth >= tabbableDate.date()
+                      ? tabbableDate
+                      : tabbableDate.clone().set('date', numberOfDaysInMonth);
+                  const isDisabled =
+                    !date?.date() ||
+                    !isRequestableDate({
+                      date: date,
+                      startDate: min,
+                      endDate: max,
+                      excludedDates,
+                      excludedDays,
+                    });
+                  const isTabbable =
+                    closestTabbableDate.format('DD/MM/YYYY') ===
+                    date?.format('DD/MM/YYYY');
```

Comment:
afaict `numberOfDaysInMonth` and `closestTabbableDate` aren't dependent on the row (ie `date` or `i`) - we should lift those out of the loop